processor/stream/benchmark_test.go

```diff
@@ -0,0 +1,68 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package stream
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io/ioutil"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/elastic/apm-server/decoder"
+	"github.com/elastic/apm-server/publish"
+	"github.com/elastic/apm-server/tests/loader"
+)
+
+func BenchmarkStreamProcessor(b *testing.B) {
+	report := func(ctx context.Context, p publish.PendingReq) error {
+		return nil
+	}
+	dir := "../testdata/intake-v2"
+	_, cwd, _, ok := runtime.Caller(0)
+	if !ok {
+		b.Error(errors.New("Could not determine test dir"))
+	}
+	files, err := ioutil.ReadDir(filepath.Join(cwd, "../..", dir))
+	if err != nil {
+		b.Error(err)
+	}
+	ctx := context.Background()
+	sp := StreamProcessor{}
```

Comment:
nitpick:

`sp := &StreamProcessor{}`

makes the call to `HandleStream`:
`sp.HandleStream` instead of `(&sp).HandleStream`